docs: Add OpenAPI specification for Curriculum API

fazzy12 opened the pull request:
This commit adds the OpenAPI specification YAML file for the Curriculum API. The YAML file includes endpoints, schemas, and descriptions for managing curriculum resources. It follows the OpenAPI 3.0.3 specification and includes paths for adding, updating, and retrieving curriculum information. 

Closes #1

FlatBallFlyer commented on docs/openapi.yaml:
```diff
@@ -1,88 +1,86 @@
     get:
-      summary: Get a list of people names and ids
-      operationId: getNames
+      summary: Get a list of all curricula
```
Likewise there will be no need to get a list of curriculum, so this endpoint can be dropped as well

FlatBallFlyer commented on docs/openapi.yaml:
```diff
@@ -1,88 +1,86 @@
-      description: Get an existing person by Id
-      operationId: gettPerson
+      summary: Get an existing curriculum
+      operationId: getCurriculum
```
Add a comment here that states if a curriculum does not exist it will be created

FlatBallFlyer commented on docs/openapi.yaml:
```diff
@@ -141,164 +106,147 @@ paths:
+          type: string
+        version:
+          description: The version of the curriculum
+          type: string
```
version properties do not be need to be exposed they are going to be deprecated soon

FlatBallFlyer commented on docs/openapi.yaml:
```diff
@@ -141,164 +106,147 @@ paths:
+        - type
+        - review
+        - roadmap
+      properties:
```
I don't think review or roadmap need to be required

FlatBallFlyer commented on docs/openapi.yaml:
```diff
@@ -141,164 +106,147 @@ paths:
+          type: string
+          enum:
+            - completed
+            - celebrated
```
Does this match the schema for the database, I don't think we need both completed and celebrated, they mean the same thing

FlatBallFlyer commented on docs/openapi.yaml:
```diff
@@ -141,164 +106,147 @@ paths:
+            - later
+        status:
+          description: Archived, Assigned, Started, Completed
+          type: string
```
The value should be listed as enum

FlatBallFlyer commented on docs/openapi.yaml:
```diff
@@ -141,164 +106,147 @@ paths:
     Config:
       type: object
       properties:
         apiVersion:
```
We will strengthen the config schema as you build the API

FlatBallFlyer reviewed (changes requested):
I think we also need to add Post patch delete endpoints for curriculum/ID/Topic. I would also appreciate it if you would put anHTML file in the docs folder similar to the one in the person API so that we can publish this swagger

FlatBallFlyer commented on docs/openapi.yaml:
```diff
@@ -1,304 +1,349 @@
           description: Validation exception
-  /api/partners/:
+
+  /api/config/:
```
It's a little bit nitpicky but can we move config and health to be the last two in points

FlatBallFlyer commented on docs/openapi.yaml:
```diff
@@ -1,304 +1,349 @@
+          description: Validation exception
+
+    patch:
+      summary: Update a topic within a curriculum
```
The patch needs a URL parameter to identify which topic to update

FlatBallFlyer commented on docs/openapi.yaml:
```diff
@@ -1,304 +1,349 @@
+          description: Validation exception
+
+    delete:
+      summary: Delete a topic from a curriculum
```
This delete operation also needs topic ID to identify the topic to delete

FlatBallFlyer reviewed (changes requested):
This is looking good, just a few minor changes here, you provided the type for those properties in the schema.

fazzy12 commented:
- Added missing descriptions for 405 responses in the openapi.yaml.
- Rearranged endpoints to position /api/config/ and /api/health/ as the last two points, improving readability.
- Enhanced the patch endpoint to include a URL parameter for specifying the topic to update.
- Modified the delete operation to incorporate a topic ID parameter for precisely identifying the topic to delete."

FlatBallFlyer reviewed (approved):
Draft to be updated when mongo schema is implemented
